x-pack/libbeat/management/managerV2.go

```diff
@@ -439,6 +451,10 @@ func (cm *BeatV2Manager) unitListen() {
 			}
 			cm.mx.Unlock()
 			cm.reload(units)
+			if cm.forceReload.Load() {
+				// Restart the debounce timer so we try to reload the inputs.
+				t.Reset(forceReloadDebounce)
```

Comment:
It'll slow down the rate the beats apply any new config it received from the agent. I The question is more, do we want it taking up to 10s to apply any new config?

Reply:
My main point here is to actually slow down the rate config changes are applied. There are some cases like Elastic-Agent + Kubernetes autodiscovery that we get a storm of updates that either conflict or lead to odd race conditions because not all inputs start running before the next call to reload.

> The question is more, do we want it taking up to 10s to apply any new config?

I would rephrase that even further: Given that those configuration changes are happening on Kibana, then need to be propagated to Fleet/Elastic-Agents, is it even noticeable a 10s delay to apply the changes/retry on failures?

I used 10s here as it is the default for the standalone config reload that works rather well.

Reply:
Yes, 10s is only for when there is an error.